### ISSUE TITLE
NavigationBar onTap ripple alignment bug

### ISSUE THREAD

**Issue by esenmx** [rydmike/flex_color_scheme#114]
Hi, using Flutter 3.7 with `flex_color_scheme: ^6.1.2`.

![flex](https://user-images.githubusercontent.com/43244505/215359994-c61e37d9-4994-4994-8606-965ac3b21afa.gif)

```dart
FlexThemeData.dark(
  scheme: FlexScheme.flutterDash,
  surfaceMode: FlexSurfaceMode.levelSurfacesLowScaffold,
  blendLevel: 15,
  subThemesData: const FlexSubThemesData(),
  keyColors: const FlexKeyColors(),
  visualDensity: FlexColorScheme.comfortablePlatformDensity,
  useMaterial3: true,
  swapLegacyOnMaterial3: true,
)
```

Thanks for your hard work!

**Comment by rydmike** [comment]
Thanks @esenmx for the issue report. 

Yes I am aware of this issue. It is a bug in Flutter 3.7, basically a regression, a bug that does not exist in Flutter 3.3 nor in Flutter master channel, since there is fix for it already in master channel.

Unfortunately that fix did not make it into Flutter 3.7 stable as cherry pick for the release. It would have needed an extra cherry pick for the 3.7 stable release, since 3.7 release had already had its code freeze when the fix was made. Now we have to wait and hope a fix for this is included in the first Flutter 3.7 hotfix (3.7.1).

The ink bug is only visible if the `NavigationBar` height is different than the default M3 height of 80dp. If you make the height lower than 80dp, the ink effect moves the diff amount from 80dp below it, if you make the height higher than 80dp, the effect moves up with same amount that it is higher than 80dp. So there is some kind of offset error on the ink ripple.

FlexColorScheme defaults to an opinionated design that is only 62 dp heighh in both M2 and M3. As a workaround, if you set its height to 80dp

```dart
  subThemesData: const FlexSubThemesData(   
    navigationBarHeight: 80.0,
  ),
```

The issue becomes invisible.

I will when I release the FCS 7.0 stable version introduce a with FCS 6.x style break, that sets the default when opting in on M3 to 80dp instead of 62dp. Which as said is the default for M3 design anyway. At least then this issue is not visible by default when opting in on M3. It will still be there when using M2 though, since I don't want to break past default for FCS when using M2. I will leave the manual workaround to set it to 80dp to be the choice of the developer when using M2. Well let's see, it depends a bit on the timeline for the fix to land.

The PR that fixes this issue can be found here:
https://github.com/flutter/flutter/issues/117420

The author of the fix @TahaTesser and @Piinks are aware that this might be a big enough issue to be included in the first hotfix to Flutter 3.7.

I will keep this issue open until there is a fix for it Flutter. I will also add it to known issues in the docs.

**Comment by rydmike** [comment]
@esenmx for information the fix to this Flutter 3.7 bug has been cherry-picked here https://github.com/flutter/flutter/issues/119554, it will thus most likely be included in the first Flutter hotfix release 3.7.1.

**Comment by rydmike** [comment]
Sadly it did not make it to 3.7.1, maybe 3.7.2 then. Yes I know, quite disappointing, but I have no control over what lands when in the framework :smile:

**Comment by rydmike** [comment]
@esenmx the Flutter SDK fix for this landed in Flutter hotfix 3.7.2. Closing this issue.
https://github.com/flutter/flutter/wiki/Hotfixes-to-the-Stable-Channel
